src/com/facebook/buck/jvm/groovy/GroovyLibraryDescription.java

```diff
@@ -0,0 +1,136 @@
+  public static final BuildRuleType TYPE = BuildRuleType.of("groovy_library");
+  public static final ImmutableSet<Flavor> SUPPORTED_FLAVORS = ImmutableSet.of();
+
+  private final GroovyBuckConfig groovyBuckConfig;
+
+  public GroovyLibraryDescription(
+      GroovyBuckConfig groovyBuckConfig) {
+    this.groovyBuckConfig = groovyBuckConfig;
+  }
+
+  @Override
+  public BuildRuleType getBuildRuleType() {
+    return TYPE;
+  }
+
+  @Override
+  public boolean hasFlavors(ImmutableSet<Flavor> flavors) {
+    return SUPPORTED_FLAVORS.containsAll(flavors);
+  }
+
+  @Override
+  public Arg createUnpopulatedConstructorArg() {
+    return new Arg();
+  }
+
+  @Override
+  public <A extends Arg> BuildRule createBuildRule(
+      TargetGraph targetGraph,
+      BuildRuleParams params,
+      BuildRuleResolver resolver,
+      A args) {
+    SourcePathResolver pathResolver = new SourcePathResolver(resolver);
+
+    BuildTarget abiJarTarget =
+        BuildTarget.builder(params.getBuildTarget())
+            .addFlavors(CalculateAbi.FLAVOR)
+            .build();
+
+    ImmutableSortedSet<BuildRule> exportedDeps = resolver.getAllRules(args.exportedDeps.get());
+    DefaultJavaLibrary defaultJavaLibrary =
+        resolver.addToIndex(
+            new DefaultJavaLibrary(
+                params.appendExtraDeps(
+                        BuildRules.getExportedRules(
+                            Iterables.concat(
+                                params.getDeclaredDeps().get(),
+                                exportedDeps,
+                                resolver.getAllRules(args.providedDeps.get())))),
+                pathResolver,
+                args.srcs.get(),
+                validateResources(
+                    pathResolver,
+                    params.getProjectFilesystem(),
+                    args.resources.get()),
+                Optional.<Path>absent(),
+                Optional.<SourcePath>absent(),
+                ImmutableList.<String>of(),
+                exportedDeps,
+                resolver.getAllRules(args.providedDeps.get()),
+                new BuildTargetSourcePath(abiJarTarget),
+                /* additionalClasspathEntries */ ImmutableSet.<Path>of(),
+                new GroovycStepFactory(
+                    groovyBuckConfig.getGroovyCompiler().get(),
+                    args.extraArguments),
+                Optional.<Path>absent(),
+                Optional.<String>absent(),
+                ImmutableSortedSet.<BuildTarget>of()));
+
+    resolver.addToIndex(
+        CalculateAbi.of(
+            abiJarTarget,
+            pathResolver,
+            params,
+            new BuildTargetSourcePath(defaultJavaLibrary.getBuildTarget())));
+
+    return defaultJavaLibrary;
+  }
+
+  @SuppressFieldNotInitialized
+  public static class Arg {
+    public Optional<ImmutableSortedSet<SourcePath>> srcs;
+    public Optional<ImmutableSortedSet<SourcePath>> resources;
+    public Optional<ImmutableList<String>> extraArguments;
```

Comment:
You should use a `MacroHandler` to support macros in here.

You can do that in a separate PR.